### PR TITLE
Normalizing feed page size

### PR DESCRIPTION
**Pull request by jeremyf**
## What type of PR is this? (check all applicable)

- [X] Refactor

## Description

This commit removes a handful of magic numbers and instead relies on a
constant.

This has a small impact in that the Basic feed will now return 50
articles instead of 25.  However, normalizing the feed pagination window
size helps reduce some oddities in reporting.

In addition, this might be something we consider giving administrators
the ability to set (with default options, because we shouldn't allow
page sizes of 10_000 as that's a massive memory hog).

## Related Tickets & Documents

Related to #14709

## QA Instructions, Screenshots, Recordings

None.

### UI accessibility concerns?

None

## Added/updated tests?

- [X] No, and this is why: I have normalized the page size and instead rely on a constant.

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [X] This change does not need to be communicated, and this is why not: this is replacing a hard-coded "magic" number with a constant

### CODE REVIEW

File: app/services/articles/feeds/basic.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 module Articles
   module Feeds
     class Basic
-      def initialize(user: nil, number_of_articles: 25, page: 1, tag: nil)
```
Comment (jeremyf) [inline app/services/articles/feeds/basic.rb:4]: This is a notable change for the `Articles::Feeds::Basic`, could someone provide context as to why I may not want this to be 50?  Perhaps @benhalpern or someone from SRE?

**Comment by jeremyf** [comment]
@benhalpern do you know why the Basic feed might be 25?  That's my most pressing concern of changing that behavior.

(Perhaps @nickytonline or @aitchiss might know a reason to not do that)

**Comment by aitchiss** [comment]
> @benhalpern do you know why the Basic feed might be 25? That's my most pressing concern of changing that behavior.
> 
> (Perhaps @nickytonline or @aitchiss might know a reason to not do that)

I'm not aware of any reason for the difference in this code, but one thing I do know is that the JS infinite scroll code defines a specific number in each page: 

https://github.com/forem/forem/blob/ca108aa9c87d2835fd2e64ab7a9520ffa47091ea/app/assets/javascripts/initializers/initScrolling.js#L279-L283

This is used for both tags feed and home feed, and doesn't seem to care whether it's the basic feed or not. So although we've changed these numbers, the frontend is forcing there to always be only 15 posts fetched/paged in.

**Comment by nickytonline** [comment]
> > @benhalpern do you know why the Basic feed might be 25? That's my most pressing concern of changing that behavior.
> > (Perhaps @nickytonline or @aitchiss might know a reason to not do that)
> 
> I'm not aware of any reason for the difference in this code, but I do know that the JS infinite scroll code defines a specific number in each page:
> 
> https://github.com/forem/forem/blob/ca108aa9c87d2835fd2e64ab7a9520ffa47091ea/app/assets/javascripts/initializers/initScrolling.js#L279-L283
> 
> This is used for both tags feed and home feed, and doesn't seem to care whether it's the basic feed or not. So although we've changed these numbers, the frontend is forcing there to always be only 15 posts fetched/paged in. As part of this change, do we want to create a constant in the JS for the `per_page` limit?

I'm not sure why that is 	`15` either in the frontend. @joshpuetz, when we worked on the feed together when we started working here, can you remember any reason why this might be have been set to `15`?

**Review by joshpuetz — COMMENTED** [review]
(never mind)

**Review by joshpuetz — APPROVED** [review]
No concerns from me 👍🏻

**Review by aitchiss — APPROVED** [review]
This code all looks good to me, so happy to approve. As I mentioned further up though I don't think this is having an impact on the number of posts shown in the logged in / paged in feed, as this all runs via the `search/feed_content` call and has a hard-coded param of 15 per page (and seems to return 25 per page if we don't include this param)

**Pull request merged**